components/Button/Button.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <button v-if="submit" type="submit" :class="classes">
+    <i v-if="icon" :class="icon"></i>
+    <slot></slot>
+    <i v-if="posticon" :class="posticon"></i>
+  </button>
+  <a v-else-if="href" :href="href" :class="classes">
+    <i v-if="icon" :class="icon"></i>
+    <slot></slot>
+    <i v-if="posticon" :class="posticon"></i>
+  </a>
+  <nuxt-link v-else-if="to" :to="to" :class="classes">
+    <i v-if="icon" :class="icon"></i>
+    <slot></slot>
+    <i v-if="posticon" :class="posticon"></i>
+  </nuxt-link>
+  <a v-else @click="$emit('click')" :class="classes">
+    <i v-if="icon" :class="icon"></i>
+    <slot></slot>
+    <i v-if="posticon" :class="posticon"></i>
+  </a>
+</template>
+
+<script>
+export default {
+  name: 'Button',
+  props: {
+    href: String,
+    to: [Object, String],
+    submit: Boolean,
+    primary: Boolean,
+    secondary: Boolean,
+    white: Boolean,
+    outline: Boolean,
+    icon: String
+  },
+  computed: {
+    classes() {
+      return {
+        'btn--primary': this.primary,
+        'btn--secondary': this.secondary,
+        'btn--white': this.white,
+        'btn--outline': this.outline
+      }
+    },
+    isExternalLink() {
+      const link = this.href || this.to
+      return link.startsWith('http')
+    },
+    posticon() {
+      if (!this.secondary) {
+        return null
+      }
+      if (this.isExternalLink) {
+        return 'far fa-external-link'
+      }
+      return 'fa fa-arrow-right'
+    }
+  }
+}
+</script>
+
+<style scoped>
+@import '../../assets/style.css';
```

Comment:
this should not be included in the component.

Either the application calling it needs to define it (or use this default one) or you can create a Theme component that actually style this.

Reply:
I will try to create a default theme